kogito-serverless-workflow/kogito-serverless-workflow-builder/src/main/java/org/kie/kogito/serverless/workflow/parser/handlers/ActionNodeUtils.java

```diff
@@ -34,16 +34,21 @@ public class ActionNodeUtils {
         return embeddedSubProcess.actionNode(context.newId()).name(functionDef.getName());
     }
 
-    public static void checkArgs(FunctionRef functionRef, String... requiredArgs) {
+    public static boolean checkArgs(ParserContext context, FunctionRef functionRef, String... requiredArgs) {
         JsonNode args = functionRef.getArguments();
+        boolean isOk = true;
         if (args == null) {
-            throw new IllegalArgumentException("Arguments cannot be null for function " + functionRef.getRefName());
-        }
-        for (String arg : requiredArgs) {
-            if (!args.has(arg)) {
-                throw new IllegalArgumentException("Missing mandatory " + arg + " argument for function " + functionRef.getRefName());
+            context.addValidationError("Arguments cannot be null for function " + functionRef.getRefName());
+            isOk = false;
+        } else {
+            for (String arg : requiredArgs) {
+                if (!args.has(arg)) {
+                    context.addValidationError("Missing mandatory " + arg + " argument for function " + functionRef.getRefName());
+                    isOk = false;
+                }
             }
         }
+        return isOk;
```

Comment:
This part without the `isOk` flag would be:

```suggestion
            context.addValidationError("Arguments cannot be null for function " + functionRef.getRefName());
            return false;
        } 
        for (String arg : requiredArgs) {
            if (!args.has(arg)) {
                context.addValidationError("Missing mandatory " + arg + " argument for function " + functionRef.getRefName());
                return false;
            }
        }
        return true;
```

Reply:
But I want to collect all missing mandatory parameters, thats the reson for the isOk